user/contactmethod/queries.sql

```diff
@@ -0,0 +1,132 @@
+-- name: AddContactMethod :exec
+INSERT INTO user_contact_methods(id, name, type, value, disabled, user_id, enable_status_updates)
+    VALUES ($1, $2, $3, $4, $5, $6, $7);
+
+-- name: UpdateContactMethod :exec
+UPDATE
+    user_contact_methods
+SET
+    name = $2,
+    disabled = $3,
+    enable_status_updates = $4
+WHERE
+    id = $1;
+
+-- name: DeleteContactMethod :exec
+DELETE FROM user_contact_methods
+WHERE id = ANY ($1::uuid[]);
+
+-- name: FindOneContactMethod :one
+SELECT
+    id,
+    name,
+    type,
+    value,
+    disabled,
+    user_id,
+    last_test_verify_at,
+    enable_status_updates,
+    pending
+FROM
+    user_contact_methods
+WHERE
+    id = $1;
+
+-- name: FindOneUpdateContactMethod :one
+SELECT
+    id,
+    name,
+    type,
+    value,
+    disabled,
+    user_id,
+    last_test_verify_at,
+    enable_status_updates,
+    pending
+FROM
+    user_contact_methods
+WHERE
+    id = $1
+FOR UPDATE;
+
+-- name: FindManyContactMethod :many
+SELECT
+    id,
+    name,
+    type,
+    value,
+    disabled,
+    user_id,
+    last_test_verify_at,
+    enable_status_updates,
+    pending
+FROM
+    user_contact_methods
+WHERE
+    id = ANY ($1::uuid[]);
+
+-- name: FindAllContactMethod :many
+SELECT
+    id,
+    name,
+    type,
+    value,
+    disabled,
+    user_id,
+    last_test_verify_at,
+    enable_status_updates,
+    pending
+FROM
+    user_contact_methods
+WHERE
+    user_id = $1;
+
+-- name: LookupUserIDContactMethod :many
+SELECT DISTINCT
+    user_id
+FROM
+    user_contact_methods
+WHERE
+    id = ANY ($1::uuid[]);
+
+-- name: EnableContactMethod :one
+UPDATE
+    user_contact_methods
+SET
+    disabled = FALSE
+WHERE
+    type = $1
+    AND value = $2
+RETURNING
+    id;
+
+-- name: MetaTVContactMethod :one
+SELECT
+    coalesce(metadata, '{}'),
+    now()::timestamptz AS now
+FROM
+    user_contact_methods
+WHERE
+    type = $1
+    AND value = $2;
+
+-- name: UpdateMetaTVContactMethod :exec
+UPDATE
+    user_contact_methods
+SET
+    metadata = jsonb_set(jsonb_set(metadata, '{CarrierV1}', @carrier_v1::jsonb), '{CarrierV1,UpdatedAt}',('"' || NOW()::timestamptz AT TIME ZONE 'UTC' || '"')::jsonb) 
```

Comment:
I see some of the queries aren't a one-to-one migration- is there any changes to the query we should know about?

It looks like this is the new replacement for:
```
setMetaTV: p.P(`
	UPDATE user_contact_methods
	SET metadata = $3
	WHERE type = $1 AND value = $2
`),
```

Reply:
Yeah this one created some issues when migrating it one-to-one. This resolves the errors by pulling only the carrierV1 field from the input instead of the whole metadata object of type `pqtype.NullRawMessage` which was causing issues. The functionality should remain the same.

Reply:
Cool just wanted to make sure since all of them are being converted here. Thank you for the clarification 🙂